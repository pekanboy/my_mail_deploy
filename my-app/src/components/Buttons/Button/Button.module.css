.container {
    width: 100px;
    height: 40px;

    background: white;
    border: 2px solid rgba(223, 226, 231, 0.89);
    /*border-left-width: 0;*/
    box-sizing: border-box;
    border-radius: 4px;

    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;

    cursor: pointer;
}

.container:hover {
    background: #E5E5E5;
}

.img {
    width: 25px;
    height: 25px;
}

.text {
    width: 50px;
    height: 20px;

    font-family: "Roboto Light", sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 14px;
    display: flex;
    align-items: center;
    justify-content: center;

    color: #000000;
}