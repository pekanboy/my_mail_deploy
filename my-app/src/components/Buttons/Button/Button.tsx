import styles from './Button.module.css';
import React, {CSSProperties} from 'react';

export interface ButtonProps {
    clickHandler: () => void
    styles?: CSSProperties
    avatar: string
    text: string
}

export const Button: React.FC<ButtonProps> = (props) => {
    return (
        <div
            className={styles.container}
            style={props.styles}
            data-name='button'
            onClick={props.clickHandler}>
            <img
                className={styles.img}
                src={props.avatar}
                alt="Изображение корзины"/>
            <span className={styles.text}>{props.text}</span>
        </div>
    );
};
