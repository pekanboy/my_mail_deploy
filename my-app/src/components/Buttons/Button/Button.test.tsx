import React, {CSSProperties} from 'react';
import {configure, shallow, ShallowWrapper} from 'enzyme';
import {Button, ButtonProps} from './Button';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const testData: ButtonProps = {
    avatar: 'http://i.pravatar.cc/300?u=1',
    text: 'Button',
    clickHandler: () => undefined,
    styles: {
        backgroundColor: 'red',
    } as CSSProperties,
};

describe('Test button component', () => {
    it('to match snapshot ', function() {
        const wrapper: ShallowWrapper = shallow(<Button {...testData} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('test callback called', () => {
        const mockClick = jest.fn();
        const data: ButtonProps = {...testData, clickHandler: mockClick};

        const wrapper: ShallowWrapper = shallow(<Button {...data} />);

        wrapper.find('div.container').at(0).simulate('click', {});

        expect(mockClick.mock.calls.length).toBe(1);
    });
});
