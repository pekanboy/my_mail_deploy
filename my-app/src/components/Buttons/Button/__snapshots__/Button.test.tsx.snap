// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button component to match snapshot  1`] = `
<div
  className="container"
  data-name="button"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <img
    alt="Изображение корзины"
    className="img"
    src="http://i.pravatar.cc/300?u=1"
  />
  <span
    className="text"
  >
    Button
  </span>
</div>
`;
